### DISCUSSION TITLE
Add support for Arch Linux

### DISCUSSION THREAD

**Discussion by connortsui20** [SeaQL/FireDBG.for.Rust#9] [Ideas]
It would be great if an Arch version could be a part of the releases!

I was able to get FireDBG working on Arch by changing the install script `install.sh` slightly:

```sh
# get_architecture || return 1
# local _arch="$RETVAL"
local _arch="x86_64-ubuntu20.04"
```

It seems like the Ubuntu 20 release tar works well with Arch, since FireDBG is running fine.

Great work all around!

**Comment by tyt2y3** [comment]
Thank you. Yeah I think for Linux the difficulty is testing whether they can work. Let's include that @billy1624

**Reply by tyt2y3** [reply]
Whats your Arch version btw?

**Reply by connortsui20** [reply]
I'm on the most recent December release of arch, with a libc++abi version 16.0 package

**Comment by billy1624** [comment]
Hey @Connortsui20, thanks for asking!! This should "fix" it https://github.com/SeaQL/FireDBG.for.Rust/commit/fe8e0b23ed3705d62ff54afb9d73db1c01a48a9f Would you like to double check?

```sh
curl https://raw.githubusercontent.com/SeaQL/FireDBG.for.Rust/main/install.sh -sSf | sh
```

**Reply by connortsui20** [reply]
Yep, it seemed like it worked!

**Comment by clemarescx** [comment]
Would it be possible to add support for Manjaro as well?

Right now the install script fails for me with `FireDBG: no precompiled binaries available for OS: manjaro` because `$_ostype` uses _ID_ instead of _ID_LIKE_ from `/etc/os-release`.

I think the inner `case` statement used to install libc++ ABI could be refactored in order to easily support more distributions, by getting the _ID_LIKE_ value from `/etc/os-release` if present, otherwise use _ID_.

In the case of Arch-based distros, _ID_LIKE_ will be `arch`, and it's very likely that in most cases they will use the Arch repository and the `pacman` package manager, and download the same `libc++abi` package - Arch Linux itself won't have an _ID_LIKE_ value but will indeed have _ID=arch_.

Unfortunately I can't test nor confirm for other distros, but I would assume it would be the same story:
- Ubuntu would have 
    - _ID=ubuntu_
    - _ID_LIKE=debian_
-  Debian
    -  _ID=debian_
- Centos 
    - _ID=centos_ 
    - _ID_LIKE="rhel fedora"_ 
- Fedora
    -  _ID="rhel fedora"_ (I guess?...)

**Reply by billy1624** [reply]
Hey @clemarescx, perhaps you can manually installing FireDBG with prebuilt binaries of `Debian 12` / `Debian 10`.

Guide: https://github.com/SeaQL/FireDBG.for.Rust/blob/main/INSTALL.md#manual-installation

If this works, we can alter the `install.sh`, make it download `Debian` prebuilt binaries for `manjaro` OS. Just like what we did for Arch Linux.

**Reply by clemarescx** [reply]
I packed the instructions from the linked guide in the following script and ran it for all releases:
```bash
#!/usr/bin/bash

fdbg_version="1.74.1"
os_version="<the target os name and version, e.g.: ubuntu20.04>"
release="x86_64-$os_version"
release_tar="$release.tar.gz"

curl -sSfL "https://github.com/SeaQL/FireDBG.for.Rust/releases/download/$fdbg_version/$release_tar" -o $release_tar

mkdir -p $release && tar xf $release_tar --strip-components 1 -C $release

mkdir -p "$HOME/.cargo/bin" && cp -r $release/* "$HOME/.cargo/bin/"

cd "$HOME/.cargo/bin/firedbg-lib/debugger-self-test"
rm -f *.firedbg.ss
firedbg run debugger_self_test --output output.firedbg.ss
firedbg-indexer --input output.firedbg.ss validate --json expected_data.json && echo "info: completed FireDBG self tests"
```
The ones that passed the self tests on Manjaro are `ubuntu20.04`, `debian10`, `centos9` and `fedora39`.
I actually tested the VSCode extension only with the `ubuntu20.04` release, seemed to work fine :slightly_smiling_face:
